deltacat/compute/compactor/README.md

```diff
@@ -0,0 +1,107 @@
+# DeltaCAT Compactor
+
+The DeltaCAT compactor provides a fast, scalable, and efficient
+Log-Structure-Merge (LSM) based Change-Data-Capture (CDC) implementation using
+Ray. It implements [The Flash Compactor Design](TheFlashCompactorDesign.pdf)
+using DeltaCAT's portable delta catalog storage APIs.
+
+## User Guide
+### Migrating to the DeltaCAT Compactor
+Migration from your old copy-on-write CDC framework to DeltaCAT is typically
+done by first running a rebase on top of your old copy-on-write compacted
+results.
+
+A rebase allows you to run compaction from source S1 to destination D on behalf
+of source S2, where S1 and S2 can either be the same or different tables.
+More specifically, it:
+1. Discards (does not read) any prior compatible round completion info and primary key indices associated with S1/S2 and D.
+2. Writes a round completion file associated with S2 and D (including a primary key index for D and an optional rebased high watermark stream position for S1).
+3. Saves and propagates the last-used rebase source across all subsequent round completion files.
+
+As part of a rebase from an alternate source or as an independent operation,
+you can optionally set a rebase source high watermark stream position that will
+be used as the starting stream position (exclusive) for the next compaction
+round.
+
+For example, a table rebase can be used to more easily transition from an old
+copy-on-write compactor to the DeltaCAT compactor by rebasing on top of the
+results of the old copy-on-write compactor.
+
+If we assume `delta_source` refers to the table that both the old compactor and
+the new compactor will read and compact deltas from, then your first call to
+`compact_partition` should look something like this:
+```python
+compact_partition(
+  source_partition_locator=old_compacted_partition,
+  compacted_partition_locator=deltacat_compacted_partition,
+  primary_keys=delta_source_primary_keys,
+  compaction_artifact_s3_bucket=deltacat_s3_bucket,
+  last_stream_position_to_compact=delta_source_last_stream_position,
+  rebase_source_partition_locator=delta_source_partition,
```

Comment:
Could we relate the partitions mentioned here to  `S1`, `S2`, `D` as stated in the above explanation? For example, something like `delta_source_partition(S2)`, `deltacat_compacted_partition(D)`.